### PR TITLE
feat(GradientBlock): implement feedback

### PR DESCRIPTION
**Pull request by Walkeezy**
https://app.clickup.com/t/8677qzbpa

- Reorder imports & cleanup some stuff
- Add color palettes to color picker
- Create helper to map app-bridge color palette to fondue color palette (use it also in color scale block)
- Adjust codemirror to match styling of code snippet block
- Adjust list order of colors to match gradient
- Adjust closing behaviour of color flyout
- Adjust styling of 'Close' button in color flyout
- Check if color is out of bound to the right and position it accordingly

### CODE REVIEW

**Review by celine-s — APPROVED** [review]
nice! Tested it and looks good 👍

File: packages/code-snippet-block/package.json [hunk 1]
```diff
@@ -31,17 +31,17 @@
         "typescript": "4.8.4"
     },
     "dependencies": {
-        "@codemirror/language": "^6.3.1",
-        "@codemirror/state": "^6.1.4",
-        "@codemirror/view": "^6.5.1",
+        "@codemirror/language": "6.3.1",
+        "@codemirror/state": "6.1.4",
```
[the remark this answers is not in the document]
Comment (Walkeezy) [reply packages/code-snippet-block/package.json:35]: I think it's just better to a have fixed version.

File: packages/gradient-block/src/GradientBlock.tsx [hunk 2]
```diff
@@ -26,11 +27,14 @@ export const GradientBlock = ({ appBridge }: BlockProps): ReactElement => {
...
     const gradientBlockRef = useRef<HTMLDivElement>(null);
-    const [addButtonPositionLeft, setAddButtonPositionLeft] = useState(0);
-    const [currentlyEditingPosition, setCurrentlyEditingPosition] = useState(0);
-    const [showAddButton, setShowAddButton] = useState(false);
-    const [showColorModal, setShowColorModal] = useState(false);
-    const [isMounted, setMounted] = useState(false);
+    const [addButtonPositionLeft, setAddButtonPositionLeft] = useState<number>(0);
+    const [currentlyEditingPosition, setCurrentlyEditingPosition] = useState<number>(0);
+    const [showAddButton, setShowAddButton] = useState<boolean>(false);
+    const [showColorModal, setShowColorModal] = useState<boolean>(false);
```
[the remark this answers is not in the document]
Comment (Walkeezy) [reply packages/gradient-block/src/GradientBlock.tsx:33]: Doesn't hurt to have them either 😅

File: packages/gradient-block/src/components/SquareBadge.tsx [hunk 3]
```diff
@@ -50,14 +53,22 @@ export const SquareBadge = ({ gradientColor, gradientOrientation, index, blockWi
         }
     };
 
+    useEffect(() => {
+        if (badgeRef.current && badgeRef.current.clientWidth + badgeRef.current.offsetLeft > blockWidth) {
+            setOutOfBounds(true);
+        }
+    }, [badgeRef, blockWidth]);
```
Comment (fulopdaniel) [inline packages/gradient-block/src/components/SquareBadge.tsx:60]: We can remove this useEffect and the outOfBounds useState, and just have a variable.

`const isOutOfBounds = !!badgeRef.current && badgeRef.current.clientWidth + badgeRef.current.offsetLeft > blockWidth`

**Review by fulopdaniel — APPROVED** [review]

**Pull request merged**